php/admin/class-admin.php

```diff
@@ -58,6 +58,11 @@ public function init() {
 		if ( $show_pro_nag && ! block_lab()->is_pro() ) {
 			$this->upgrade = new Upgrade();
 			block_lab()->register_component( $this->upgrade );
+		} else {
+			$page = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING );
+			if ( 'block-lab-pro' === $page ) {
+				$this->settings_redirect();
+			}
```

Comment:
Maybe all of this `else` block could be inside `settings_redirect`, as it looks like that method is only used once. If you agree, maybe another name would be good, like `maybe_settings_redirect()`.

```php
} else {
        $this->maybe_settings_redirect();
}
```

Reply:
Great idea. Done in https://github.com/getblocklab/block-lab/pull/327/commits/383bfe4a299e91cb53cbef85c654ff7e6e2ace93.